src/tap_reporter.wren

```diff
@@ -0,0 +1,53 @@
+import "./capabilities" for Capabilities
+import "./stacktrace_report" for StackTraceReport
+import "./expect" for Expect
+
+StackTraceReport.withoutColors
+Expect.withoutColors
+
+class TapReporter {
+    construct new(name) {
+        _name = name
+        _fail = _skip = _success = _number = 0
+        _plan = false
```

Comment:
What is this?  I'm not sure I find the naming clear.

Reply:
The "plan" is a TAP concept: https://testanything.org/tap-version-13-specification.html#the-plan

Before you start testing, you might know how many tests you _plan_ to run. But you might not know.

Reply:
Ok, but I don't understand what the boolean is tracking... it is set to true when we start but then it only outputs the plan at the end if it's false?  When would it be false?  `done` is called after `start`, yes?

Reply:
`start(num_tests)` sets it to true. `start()` does not.

Reply:
Ok, now I finally understand.  What about:

```
_number_tests_to_run = null # all
_number_tests_to_run = 5 
show_plan_ {} => "before" || "after"
```

I think that would make what is happening and why much clearer?